models/classes/resources/Service/InstanceCopier.php

```
@@ -97,7 +97,8 @@ public function copy(
     private function doCopy(
         core_kernel_classes_Resource $instance,
         core_kernel_classes_Class $destinationClass,
-        bool $keepOriginalPermissions = true
+        bool $keepOriginalPermissions = true,
```

PHPMD (2.15.0)

The method doCopy has a boolean flag argument $keepOriginalPermissions, which is a certain sign of a Single Responsibility Principle violation. (Clean Code Rules)

(BooleanArgumentFlag)